src/contexts/quiz-mode/application/use-cases/create-quiz-from-material.ts

```diff
@@ -20,7 +20,23 @@ export class CreateQuizFromMaterialUseCase implements CreateQuizFromMaterial {
         };
     }
 
-    private extractContent(project: Project): string {
-        return `Create quiz about Poland's geography - ${project.title}`;
+    private async extractContent(project: Project, materialRepo: MaterialRepo): Promise<string> {
+        const materialIds = project.materialIds;
+        const allMaterials = await materialRepo.getAll();
+        const projectMaterials = allMaterials.filter(material => materialIds.includes(material.id));
+        
+        if (projectMaterials.length === 0) {
+            return project.title;
+        }
+
+        const contentParts = projectMaterials.map(material => {
+            const title = material.title;
+            // We have differnt types of content, do we have an idea how we want to parse them so that
+            // it can be send to the llms?
```

Comment:
yes, but for now we assume the pdfs are small enough for that. 1.0.0 should do some kind of chunking and quiz on limited context or have a rag pipeline